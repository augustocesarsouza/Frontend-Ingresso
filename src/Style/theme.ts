export const theme = {
  colors: {
    primaryColor: '#333', // meio-escuro
    secondaryColor: '#2634a2', // azul
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },
  font: {
    family: {
      default: "'Roboto'",
      secondary: "'Arial'",
    },
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};

// export interface Theme {
//   colors: {
//     primaryColor: string;
//     secondaryColor: string;
//     white: string;
//     mediumGray: string;
//   },
//   font: {
//     family: {
//       default: string;
//       secondary: string;
//     },
//     sizes: {
//       xsmall: string;
//       small: string;
//       medium: string;
//       large: string;
//       xlarge: string;
//       xxlarge: string;
//       huge: string;
//       xhuge: string;
//     },
//   },
//   media: {
//     lteMedium: string;
//   },
//   spacings: {
//     xsmall: string;
//     small: string;
//     medium: string;
//     large: string;
//     xlarge: string;
//     xxlarge: string;
//     huge: string;
//     xhuge: string;
//   },
// }
