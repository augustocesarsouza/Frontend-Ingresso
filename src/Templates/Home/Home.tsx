import Header from '../Header/Header';
import { useLocation } from 'react-router-dom';
import { useState, useEffect, createContext } from 'react';
import { ObjUser } from '../../Components/LoginComponents/SectionFirst/Form/Form';

export interface ContextHomeProps {
  userObj: ObjUser;
  setUserObj: React.Dispatch<React.SetStateAction<ObjUser | null>>;
}

export const ContextHome = createContext<ContextHomeProps | null>(null);

const Home = () => {
  const [userObj, setUserObj] = useState<ObjUser | null>(null);
  const location = useLocation();

  useEffect(() => {
    if (location.state === null) return;
    const user = location.state.user;

    setUserObj(user);
  }, [location]);

  return (
    <ContextHome.Provider
      value={{
        userObj,
        setUserObj,
      }}
    >
      <Header />
    </ContextHome.Provider>
  );
};

export default Home;
