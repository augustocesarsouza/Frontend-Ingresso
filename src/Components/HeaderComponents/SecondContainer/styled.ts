import styled, { css } from 'styled-components';
import { Theme } from '../../../Style/theme';

export const SecondContainer = styled.div`
  display: flex;
  width: 50%;
  align-items: center;
  justify-content: flex-end;
  gap: 30px;
`

export const ContainerLoupe = styled.div`
  cursor: pointer;
  ${({theme}: { theme: Theme }) => css`
    svg {
      width: ${theme.svg.width};
      fill: ${theme.svg.fill};
    }

    &:hover {
      svg {
        fill: ${theme.svg.fillhover};
      }
    }
  `}
`

export const ContainerUserImgSvg = styled.div`
  display: flex;
  gap: 12px;
  position: relative;
`

export const ContainerQuestion = styled.div`
  margin-right: 80px;
  cursor: pointer;

  ${({theme}: { theme: Theme }) => css`
    svg {
      width: ${theme.svg.width};
      fill: ${theme.svg.fill};
    }

    &:hover {
      svg {
        fill: ${theme.svg.fillhover};
      }
    }
  `}
`

export const P = styled.p`
  color: #ffffff;
  font-weight: 600;
  user-select: none;
`

export const ContainerNavMainUser = styled.div`
  position: absolute;
  left: -285px;
  top: 50px;
  background-color: #191919;
  width: 51rem;
  height: 30rem;
  display: flex;
  padding: 24px 0px;

  svg {
    position: relative;
    color: #191919;
    font-size: 39px;
    left: 290px;
    bottom: 45px;
  }
`

export const ContainerNav = styled.div`
  width: 100%;
  gap: 25px;
  display: flex;
  flex-direction: column;
  user-select: none;
`

export const ContainerLineWhite = styled.div`
  border-bottom: 1px solid #ffffff5e;
  width: 95%;
`

export const Nav = styled.nav`
  display: flex;
  flex-direction: column;
  gap: 25px;
`

export const Link = styled.a`
  font-weight: 500;
  color: #0d8fff;
  cursor: pointer;

  &:hover {
    color: #0a71c9;
  }
`