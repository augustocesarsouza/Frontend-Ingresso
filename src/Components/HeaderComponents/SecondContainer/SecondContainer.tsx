import QuestionMarkSvg from '../../../Svg/QuestionMarkSvg';
import LoupeSvg from '../../../Svg/loupeSvg';
import * as Styled from './styled';
import { useState, useEffect, useContext } from 'react';
import EnterOrRegister from '../EnterOrRegister/EnterOrRegister';
import { ContextHome, ContextHomeProps } from '../../../Templates/Home/Home';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretUp } from '@fortawesome/free-solid-svg-icons';
import InfoLoggedUser from '../InfoLoggedUser/InfoLoggedUser';
import { useNavigate } from 'react-router-dom';

const SecondContainer = () => {
  const [enterOrRegister, setEnterOrRegister] = useState(false);
  const [enterMouseDivUserImgSvg, setEnterMouseDivUserImgSvg] = useState(false);
  const [userName, setUserName] = useState('');
  const nav = useNavigate();

  const useContextHome = useContext<ContextHomeProps | null>(ContextHome);

  const handleEnterOrRegister = () => {
    setEnterOrRegister((prev) => !prev);
  };

  const documentEvent = () => {
    setEnterOrRegister(false);
  };

  useEffect(() => {
    if (enterMouseDivUserImgSvg) return;
    document.addEventListener('click', documentEvent);

    return () => {
      document.removeEventListener('click', documentEvent);
    };
  }, [enterMouseDivUserImgSvg]);

  useEffect(() => {
    if (useContextHome.userObj === null) return;

    const {
      userObj: { name },
    } = useContextHome;
    setUserName(name);
  }, [useContextHome]);

  const handleEnter = () => {
    setEnterMouseDivUserImgSvg(true);
  };

  const handleLeave = () => {
    setEnterMouseDivUserImgSvg(false);
  };

  const [openInfoUser, setOpenInfoUser] = useState(false);

  const handleClickUserName = () => {
    setOpenInfoUser((prev) => !prev);
  };

  const handleMyOrders = () => {
    if (useContextHome.userObj === null) return;
    nav('/minha-conta/meus-pedidos', { state: { user: useContextHome.userObj } });
  };

  const handlePersonalData = () => {};

  const handlePaymentMethods = () => {};

  const handleLoggingOut = () => {
    if (useContextHome.userObj === null) return;

    const { setUserObj } = useContextHome;

    setOpenInfoUser(false);
    nav('/', { state: { user: null } });
    setUserName('');
    setUserObj(null);
  };

  return (
    <Styled.SecondContainer>
      <Styled.ContainerLoupe data-testid="container-loupe">
        <LoupeSvg />
      </Styled.ContainerLoupe>
      <Styled.ContainerUserImgSvg onMouseEnter={handleEnter} onMouseLeave={handleLeave}>
        <InfoLoggedUser
          userName={userName}
          handleEnterOrRegister={handleEnterOrRegister}
          handleClickUserName={handleClickUserName}
        />
        <EnterOrRegister enterOrRegister={enterOrRegister} />
        {openInfoUser && (
          <Styled.ContainerNavMainUser>
            <FontAwesomeIcon icon={faCaretUp} />
            <Styled.ContainerNav>
              <Styled.P>Ol√°, {userName}</Styled.P>
              <Styled.Nav>
                <Styled.Link onClick={handleMyOrders}>Meus Pedidos</Styled.Link>
                <Styled.Link onClick={handlePersonalData}>Dados Pessoais</Styled.Link>
                <Styled.Link onClick={handlePaymentMethods}>Formas de Pagamento</Styled.Link>

                <Styled.ContainerLineWhite></Styled.ContainerLineWhite>
                <Styled.Link onClick={handleLoggingOut}>Sair da Conta</Styled.Link>
              </Styled.Nav>
            </Styled.ContainerNav>
          </Styled.ContainerNavMainUser>
        )}
      </Styled.ContainerUserImgSvg>
      <Styled.ContainerQuestion data-testid="container-question-mark">
        <QuestionMarkSvg />
      </Styled.ContainerQuestion>
    </Styled.SecondContainer>
  );
};

export default SecondContainer;
