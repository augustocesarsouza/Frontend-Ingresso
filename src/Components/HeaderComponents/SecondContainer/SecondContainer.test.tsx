import { screen } from '@testing-library/react';
import SecondContainer from './SecondContainer';
import { RenderTheme } from '../../../Style/RenderTheme';
import { MemoryRouter } from 'react-router-dom';
import { ContextHome } from '../../../Templates/Home/Home';

describe('SecondContainer', () => {
  let userObj = {
    id: '59a889f9-2695-45bf-b846-2b207e1b5539',
    name: 'Augusto',
    email: 'augustocesarsantana90@gmail.com',
  };

  const setUserObj = null;

  test('should have justify-content: flex-end', () => {
    RenderTheme(
      <MemoryRouter>
        <ContextHome.Provider value={{ userObj, setUserObj }}>
          <SecondContainer />
        </ContextHome.Provider>
      </MemoryRouter>
    );

    const containerLocation = screen.getByTestId('container-loupe').parentElement;

    expect(containerLocation).toHaveStyle({
      'justify-content': 'flex-end',
    });
  });

  test('should render svg Loupe', () => {
    RenderTheme(
      <MemoryRouter>
        <ContextHome.Provider value={{ userObj, setUserObj }}>
          <SecondContainer />
        </ContextHome.Provider>
      </MemoryRouter>
    );

    const containerLocation = screen.getByTestId('container-loupe');
    const svgLoupe = containerLocation.firstChild as SVGElement;

    expect(svgLoupe).toBeInTheDocument();
  });

  test('should render svg questionMarkSvg', () => {
    RenderTheme(
      <MemoryRouter>
        <ContextHome.Provider value={{ userObj, setUserObj }}>
          <SecondContainer />
        </ContextHome.Provider>
      </MemoryRouter>
    );

    const containerQuestionMark = screen.getByTestId('container-question-mark');
    const svgQuestionMark = containerQuestionMark.firstChild as SVGElement;

    expect(svgQuestionMark).toBeInTheDocument();
  });

  test('should matchSnapshot', () => {
    const { container } = RenderTheme(
      <MemoryRouter>
        <ContextHome.Provider value={{ userObj, setUserObj }}>
          <SecondContainer />
        </ContextHome.Provider>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    //Se mecher no codigo na logica de negocio atualizar com "npx jest --updateSnapshot"
  });
});
