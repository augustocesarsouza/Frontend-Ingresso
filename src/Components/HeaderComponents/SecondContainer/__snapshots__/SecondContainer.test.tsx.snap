// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondContainer should matchSnapshot 1`] = `
.c0 {
  display: flex;
  width: 50%;
  align-items: center;
  justify-content: flex-end;
  gap: 30px;
}

.c1 {
  cursor: pointer;
}

.c1 svg {
  width: 3.5rem;
  fill: rgb(231 235 248);
}

.c1:hover svg {
  fill: rgb(221 227 248 / 85%);
}

.c2 {
  display: flex;
  gap: 12px;
  position: relative;
}

.c6 {
  margin-right: 80px;
  cursor: pointer;
}

.c6 svg {
  width: 3.5rem;
  fill: rgb(231 235 248);
}

.c6:hover svg {
  fill: rgb(221 227 248 / 85%);
}

.c3 {
  cursor: pointer;
}

.c3 svg {
  width: 3.5rem;
  fill: rgb(231 235 248);
}

.c3:hover svg {
  fill: rgb(221 227 248 / 85%);
}

.c4 {
  color: #ffc811;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  user-select: none;
}

.c5 {
  width: 9rem;
  font-size: 14px;
  cursor: pointer;
  color: rgb(231 235 248);
  user-select: none;
}

.c5:hover {
  color: rgb(221 227 248 / 85%);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="container-loupe"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.75 2.75a6.625 6.625 0 0 1 4.99 10.983c.039.027.077.06.113.095l5.569 5.57c.341.341.36.875.043 1.192-.317.318-.851.298-1.193-.043l-5.569-5.57a.939.939 0 0 1-.142-.182A6.625 6.625 0 1 1 9.75 2.75zm0 1.625a5 5 0 1 0 0 10 5 5 0 0 0 0-10zm-3.417 3.74c.206.019.36.2.341.406-.184 2.152.69 3.352 2.703 3.727a.375.375 0 0 1-.138.737c-2.398-.447-3.529-1.998-3.312-4.528a.375.375 0 0 1 .406-.341z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          class="user-logo-svg"
          viewBox="0 0 64 64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 6.667c13.991 0 25.333 11.342 25.333 25.333 0 13.991-11.342 25.333-25.333 25.333C18.009 57.333 6.667 45.991 6.667 32 6.667 18.009 18.009 6.667 32 6.667zm0 14.75c-4.631 0-5.948 2.607-5.058 8.026.04.24.034.486-.014.724-.362 1.768.122 3.687 1.238 5.742.32.588 1.623 2.583 1.75 3.008a2.001 2.001 0 0 1-1.298 2.476c-5.486 1.784-10.476 3.566-12.52 4.555 3.883 4.52 9.642 7.385 16.069 7.385 6.318 0 11.99-2.768 15.868-7.158l-12.26-4.819a2 2 0 0 1-.986-2.907c1.344-2.202 2.24-4.029 2.688-5.451.367-1.164.395-1.915.238-2.246a1.999 1.999 0 0 1-.19-.966c.373-6.9-.518-8.37-5.525-8.37zM32.167 11C20.477 11 11 20.477 11 32.167c0 3.817 1.01 7.398 2.778 10.49 2.005-1.05 6.496-2.693 11.192-4.272-.107-.184-.212-.37-.318-.566-1.458-2.683-2.165-5.351-1.713-8.079-.47-3.134-.282-5.824.804-7.973 1.428-2.83 4.245-4.35 8.257-4.35 4.048 0 6.728 1.094 8.228 3.568 1.199 1.976 1.506 4.493 1.313 8.581.724 2.175-.053 5.075-2.141 8.902l11.056 4.36a21.054 21.054 0 0 0 2.877-10.661C53.333 20.477 43.857 11 32.167 11z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class=""
      >
        <p
          class="c4"
        >
          Ol√°, 
          Augusto
        </p>
        <span
          class="c5"
        >
          Meus pedidos
        </span>
      </div>
    </div>
    <div
      class="c6"
      data-testid="container-question-mark"
    >
      <svg
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 6.667c13.991 0 25.333 11.342 25.333 25.333 0 13.991-11.342 25.333-25.333 25.333C18.009 57.333 6.667 45.991 6.667 32 6.667 18.009 18.009 6.667 32 6.667zM32.167 11C20.477 11 11 20.477 11 32.167c0 11.69 9.477 21.166 21.167 21.166 11.69 0 21.166-9.476 21.166-21.166S43.857 11 32.167 11zm0 31.667A2.853 2.853 0 0 1 35 45.5c0 1.519-1.298 2.833-2.833 2.833-1.535 0-2.834-1.314-2.834-2.833a2.854 2.854 0 0 1 2.834-2.833zM32 17.333c5.46 0 9.332 3.812 9.332 9.417 0 2.767-1.396 5.623-4.24 7.56a22.575 22.575 0 0 1-2.84 1.631l-.049.024v2.702a2 2 0 0 1-4 0v-4a2 2 0 0 1 1.278-1.865c.037-.015.125-.05.257-.107a19.07 19.07 0 0 0 3.102-1.692c.691-.466 2.492-1.29 2.492-4.253s-1.559-5.416-5.25-5.416c-3.972 0-5.416 2.323-5.416 5.5 0 1.104-1.23 2-2.334 2s-2-.896-2-2c0-5.47 4.079-9.5 9.668-9.5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
